lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -231,7 +231,7 @@ defmodule Mix.Compilers.Elixir do
   end
 
   defp detect_kind(module) do
-    impl = Module.get_attribute(module, :impl)
```

Comment:
For peace of mind, maybe it's a good idea to rename this variable as well (so we don't have this `impl` raising questions here). Wdyt?